File: src/api/tokenStore.js
import AsyncStorage from '@react-native-async-storage/async-storage';
import * as SecureStore from 'expo-secure-store';

const JWT_KEY = 'grayman.jwt';
const WORKER_ID_KEY = 'grayman.workerID';

let cachedToken = null;

export const tokenStore = {
  get token() {
    return cachedToken;
  },

  get workerID() {
    return AsyncStorage.getItem(WORKER_ID_KEY);
  },

  async load() {
    try {
      cachedToken = await SecureStore.getItemAsync(JWT_KEY);
    } catch {
      cachedToken = await AsyncStorage.getItem(JWT_KEY);
    }
    return cachedToken;
  },

  async save(token, workerId) {
    cachedToken = token;
    try {
      await SecureStore.setItemAsync(JWT_KEY, token);
    } catch {
      await AsyncStorage.setItem(JWT_KEY, token);
    }
    if (workerId) {
      await AsyncStorage.setItem(WORKER_ID_KEY, workerId);
    }
  },

  async clear() {
    cachedToken = null;
    try {
      await SecureStore.deleteItemAsync(JWT_KEY);
    } catch { /* ignore */ }
    await AsyncStorage.multiRemove([JWT_KEY, WORKER_ID_KEY]);
  },

  async getWorkerID() {
    return AsyncStorage.getItem(WORKER_ID_KEY);
  },
};

File: flutter_app/lib/core/networking/token_store.dart
import 'package:flutter_secure_storage/flutter_secure_storage.dart';
import 'package:shared_preferences/shared_preferences.dart';

/// Mirrors iOS TokenStore — JWT in secure storage, worker ID in prefs.
class TokenStore {
  TokenStore._();
  static final TokenStore instance = TokenStore._();

  static const _jwtKey = 'jwt';
  static const _workerIdKey = 'grayman.workerID';
  final _secure = const FlutterSecureStorage();

  String? _cached;

  String? get token => _cached;
  String? workerId;

  Future<void> load() async {
    _cached = await _secure.read(key: _jwtKey);
    final prefs = await SharedPreferences.getInstance();
    workerId = prefs.getString(_workerIdKey);
  }

  Future<void> save(String token, {String? workerId}) async {
    _cached = token;
    await _secure.write(key: _jwtKey, value: token);
    if (workerId != null) {
      this.workerId = workerId;
      final prefs = await SharedPreferences.getInstance();
      await prefs.setString(_workerIdKey, workerId);
    }
  }

  Future<void> clear() async {
    _cached = null;
    workerId = null;
    await _secure.delete(key: _jwtKey);
    final prefs = await SharedPreferences.getInstance();
    await prefs.remove(_workerIdKey);
  }
}
